Engine/src/Engine/Scene.cpp

```diff
@@ -56,6 +77,23 @@ namespace Engine
 	void Scene::onRuntimeUpdate()
 	{
 		renderScene();
+
+		//start the UI
+		UserInterface::startUI();
```

Comment:
I would say that we should have a separate onEditorRuntime to handle all updates related to editor specifically

Reply:
We can then use the editor window to actually show the render of the game, or just use a regular window if the edit is not launched (such as a game executable for example)

Reply:
I'm not sure I understand. The UI is part of the core engine, there is no editor window per-se. Where do you want this onEditorRuntime to be?

Reply:
Basically we want to separate editor update and game update. That way when we build a game executable, we are not also building the editor as well (just the game window). By separating these onUpdate functions, we can make that process a lot easier later on.

Reply:
Created a RenderUI()